sympy/parsing/c/c_parser.py

```diff
@@ -270,8 +279,6 @@ def transform_var_decl(self, node):
                     or child.kind == cin.CursorKind.UNEXPOSED_EXPR):
 
                     if node.type.kind in self._data_types["int"]:
-                        type = self._data_types["int"][node.type.kind]
-
                         # when only one decl_ref_expr is assigned
                         # e.g., int b = a;
                         if isinstance(val, str):
```

Comment:
There is still a bit of redundancy in the code.

I think it might be better to make these the outer if statements as `value` is the same in case of strings and the else statements. The other if statements can have the `if node.type.kind in self._data_types["int"]` as nested statements accordingly.

Reply:
Looking into it, how those redundancies can be removed

Reply:
Hey, don't you think, in that case, we will have to write `if node.type.kind in self._data_types["blah"]:` again inside those two if statements, resulting in a redundancy again

Reply:
See:
``` python
                    if isinstance(val, str):
                        value = Symbol(val)
                    elif isinstance(val, bool):
                        if node.type.kind in self._data_types["int"]:
                            value = Integer(0) if val == False else Integer(1)
                        elif node.type.kind in self._data_types["float"]:
                            value = Float(0.0) if val == False else Float(1.0)
                        elif node.type.kind in self._data_types["bool"]:
                            value = sympify(val)
                    elif isinstance(val, (Integer, int, Float, float)):
                        if node.type.kind in self._data_types["int"]:
                            value = Integer(val)
                        elif node.type.kind in self._data_types["float"]:
                            value = Float(val)
                        elif node.type.kind in self._data_types["bool"]:
                            value = sympify(bool(val))
                    else:
                        value = val
```

Reply:
Itis also less code than previous and nothing is being repeated at least. The if statements are repeated but they have different statements